torch/lib/c10d/test/TCPStoreTest.cpp

```diff
@@ -0,0 +1,87 @@
+#include "StoreTestCommon.hpp"
+#include "TcpStore.hpp"
+
+#include <thread>
+#include <iostream>
+#include <cstdlib>
+
+
+int main(int argc, char** argv) {
+
+  // Master store
+  c10d::TCPStore masterStore("127.0.0.1", 29500, true);
+
+  // Basic set/get on the master store
+  c10d::test::set(masterStore, "key0", "value0");
+  c10d::test::set(masterStore, "key1", "value1");
+  c10d::test::set(masterStore, "key2", "value2");
+  c10d::test::check(masterStore, "key0", "value0");
+  c10d::test::check(masterStore, "key1", "value1");
+  c10d::test::check(masterStore, "key2", "value2");
+
+  // Hammer on TCPStore
+  std::vector<std::thread> threads;
+  const auto numThreads = 16;
+  const auto numIterations = 1000;
+  c10d::test::Semaphore sem1, sem2;
+
+  // Each thread will have a slave store to send/recv data
+  std::vector<std::unique_ptr<c10d::TCPStore>> slaveStores;
```

Comment:
They are not slave stores (replicas or something). Can we stick to daemon/server and worker/client?

Reply:
done